scio-core/src/main/scala/com/spotify/scio/sql/Sql.scala

```diff
@@ -245,8 +242,8 @@ object Queries {
 
   private[this] def printInferred(inferredSchemas: List[(String, BSchema)]): String =
     inferredSchemas
-      .map {
-        case (name, schema) =>
+      .collect {
+        case (name: String, schema: BSchema) =>
```

Comment:
Why the type matching ? The type are concrete. `inferredSchemas: List[(String, BSchema)]`

Reply:
yeah feels meh, but `BSchema` is a java type it can be null and without type it will match. 

```scala
scala> List[java.lang.Integer](null).collect { case x => x }
res3: List[Integer] = List(null)

scala> List[java.lang.Integer](null).collect { case x: java.lang.Integer => x }
res4: List[Integer] = List()
```

¯\_(ツ)_/¯

Reply:
I would rather null check and tell in the message that the schema was not indfred properly than just swallow it.

Reply:
we can do that!